Zend/zend_gc.c

```diff
@@ -577,25 +625,38 @@ static void gc_mark_grey(zend_refcounted *ref)
 static void gc_mark_roots(void)
 {
 	gc_root_buffer *current = GC_G(roots).next;
+	zend_refcounted *ref;
+	gc_refcounted_stack *stack = gc_refcounted_stack_init(GC_REFCOUNTED_STACK_INITIAL_SIZE);
```

Comment:
Rather than creating a separate stack in each of these functions, I think it would be better to create one stack in zend_gc_collect_cycles and then pass it to the individual functions. This way we can avoid going through the stack growing each time.

Reply:
Right. But I'm going to use one stack for gc_mark_grey, gc_scan and gc_collect_white and another one for gc_scan_black as this last one runs during gc_scan.